woocommerce/wc-admin
# Task list - add a shortcut back to store setup

Fixes #4592 

This adds the functionality to support a store setup tab displayed in ActivityPanel. The tab currently just redirects to the home page, rather than implementing the nice-to-haves of displaying the task list in the panel. The reason for this, is that I found it would require significant refactoring of the task list to support this currently.

As part of this I have also refactored rendering of tabs inside <ActivityPanel> into 2 new functional components: `<Tabs>` and `<Tab>`. The reason for this is to decompose large untestable components, test them and by virtue of being decomposed they will also be more reusable in future. Ideally <ActivityPanel> would have been refactored to a functional component as well, but it would have been too large a task.

**The logic for showing the setup progress tab is something I'd be keen to have checked. At the moment it only displays if:**

* You're not on the home screen
* You haven't completed the setup task list
* You're not working on one of the setup tasks

### Accessibility




### Screenshots

<img width="1294" alt="Screenshot 2020-07-23 at 10 40 43 AM" src="https://user-images.githubusercontent.com/1281828/88236384-fe4dd900-ccd0-11ea-92e2-058c2ee15f12.png">

### Detailed test instructions:

#### Test the ActivityPanel refactor

Because there has been some refactoring of the ActivityPanel in general it would be really great to smoke test its general functionality such as:

* click outside close the menu
* switching between different panels (note the animations still work)
* view page at mobile size and check the mobile menu toggle still works.

#### Test the Setup Progress tab

Testing the setup progress tab will require having a site with some setup tasks not completed. 

* Visit the home screen and confirm its not shown there.
* Go to a setup task and confirm its not shown there
* Go to other menu items in WooCommerce and confirm its shown there.
* Click the tab. You should be taken back to the home screen
